improve three debug messages

bernt-matthias opened the pull request:
## What did you do? 

Small change to ~two~ three (debug) messages

## Why did you make this change?

tool evaluation: for tools with many outputs its helpful to know which datasets and collections have been created.

drmaa: for debugging its more convenient to see the command that could be copy pasted

filter errors: debug is not shown in planemo output where it would be really helpful. 
- [ ] actually I would like to reraise here .. the try except block is there from the beginning https://github.com/galaxyproject/galaxy/commit/89ff0efe84c99fcaa3fb8d7eab38be26088b8c04 ... maybe this is now a feature .. 

## How to test the changes? 
(select the most appropriate option; if the latter, provide steps for testing below)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

## For UI Components
- [ ] I've included a screenshot of the changes

bernt-matthias committed:
improve two debug messages

tool evaluation: for tools with many outputs its helpful to know which datasets
and collections have been created.

drmaa: for debugging its more convenient to see the command that could be copy
pasted

bernt-matthias committed:
promote filter faiure message

debug messages are not shown in planemo output where
the message can be really helpful.

Co-authored-by: David López <46503462+davelopez@users.noreply.github.com>

davelopez commented on lib/galaxy/tools/actions/__init__.py:
```diff
@@ -949,7 +949,7 @@ def filter_output(tool, output, incoming):
                 return True  # do not create this dataset
         except Exception as e:
-            log.debug('Tool %s output %s: dataset output filter (%s) failed: %s' % (tool.id, output.name, filter.text, e))
+            log.info('Tool %s output %s: dataset output filter (%s) failed: %s' % (tool.id, output.name, filter.text, e))
```
Why not use f-string here too? :)
```suggestion
            log.info(f'Tool {tool.id} output {output.name}: dataset output filter ({filter.text}) failed: {e}')
```

mvdbeek commented on lib/galaxy/tools/actions/__init__.py:
```diff
@@ -949,7 +949,7 @@ def filter_output(tool, output, incoming):
                 return True  # do not create this dataset
         except Exception as e:
-            log.debug('Tool %s output %s: dataset output filter (%s) failed: %s' % (tool.id, output.name, filter.text, e))
+            log.info(f'Tool {tool.id} output {output.name}: dataset output filter ({filter.text}) failed: {e}')
```
This should be at the debug level, that's going to spam the logs even if everything is alright. The log level for this module needs to be changed in planemo.

bernt-matthias replied:
Alright, then I revert this. Can you tell me an example when this is triggered and you do not want to see the message?

bernt-matthias replied:
Would this be done here: https://github.com/galaxyproject/planemo/blob/f5b2a4aa2d0f93c1152adddde0ac61f037fe64ad/planemo/galaxy/config.py#L130?

mvdbeek replied:
That's the rough idea, would be awesome if we can tune it there.

bernt-matthias replied:
Maybe like so: https://github.com/galaxyproject/planemo/pull/1165

bernt-matthias committed:
Revert "promote filter faiure message"

This reverts commit 140ca7404cbca12d2c665ca8742d420b7be2fc2b.

bernt-matthias committed:
uvicorn: do not output access info messages

one could also pass `log_level="warning"` but this swallows some useful
messages at startup and shutdown, ie.

```
INFO:     Started server process [1394761]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8296 (Press CTRL+C to quit)
...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [1394761]
```

mvdbeek commented on lib/galaxy_test/driver/driver_util.py:
```diff
@@ -563,7 +563,7 @@ def uvicorn_serve(app, port, host=None):
     from uvicorn.server import Server
     from uvicorn.config import Config
-    config = Config(app, host=host, port=int(port))
+    config = Config(app, host=host, port=int(port), access_log=False)
```
Nice! But for Galaxy tests the access logs are useful sometimes. Maybe that could be controlled with an environment variable ? `GALAXY_TEST_ACCESS_LOG` ? Then planemo can set this but we keep the access logs in the CI.

bernt-matthias replied:
sure. is there an example where an env var is used somewhere in the code base?

mvdbeek replied:
https://github.com/mvdbeek/galaxy/blob/20d92afc5cae22d22528bbfd84eeadc94939efc5/lib/galaxy_test/driver/driver_util.py#L983

bernt-matthias committed:
control uvicorn access log with env variable

mvdbeek commented on lib/galaxy_test/driver/driver_util.py:
```diff
@@ -563,7 +563,14 @@ def uvicorn_serve(app, port, host=None):
     from uvicorn.config import Config
-    config = Config(app, host=host, port=int(port))
+
+    access_log = os.environ.get('GALAXY_UVICORN_ACCESS_LOG', None)
```
```suggestion
    access_log = os.environ.get('GALAXY_TEST_UVICORN_ACCESS_LOG', None)
```
All the variables that control Galaxy via the test driver use GALAXY_TEST_* prefixes.

bernt-matthias committed:
rename environment variable

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

bernt-matthias committed:
doc

mvdbeek commented on run_tests.sh:
```diff
@@ -246,6 +246,7 @@ TOOL_SHED_TEST_TMP_DIR          Defaults to random /tmp directory - place for
                                 tool shed test server files to be placed.
 TOOL_SHED_TEST_OMIT_GALAXY      Do not launch a Galaxy server for tool shed
                                 testing.
+GALAXY_TEST_UVICORN_ACCESS_LOG  Do not output uvicorn access messages to log if non-enmpty
```
```suggestion
GALAXY_TEST_DISABLE_ACCESS_LOG  Do not log access messages
```
I think we don't need to be that specific here. We won't disable paster or uwsgi access logs, but those aren't used by default. Also by naming it that way we don't need to parse the value of `GALAXY_TEST_DISABLE_ACCESS_LOG`.

mvdbeek commented on lib/galaxy_test/driver/driver_util.py:
```diff
@@ -563,7 +563,14 @@ def uvicorn_serve(app, port, host=None):
+    else:
+        eal = True
+
+    config = Config(app, host=host, port=int(port), access_log=eal)
```
```suggestion
    access_log = False if 'GALAXY_TEST_DISABLE_ACCESS_LOG' in os.environ else True
    config = Config(app, host=host, port=int(port), access_log=access_log)
```

bernt-matthias committed:
Apply suggestions from code review

- fix and simplify reading of env variable
- shorten doc

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>
